src/test/java/htsjdk/samtools/CRAMIteratorTest.java

```diff
@@ -0,0 +1,50 @@
+package htsjdk.samtools;
+
+import htsjdk.HtsjdkTest;
+import htsjdk.samtools.cram.ref.ReferenceSource;
+import htsjdk.samtools.seekablestream.SeekableStream;
+import org.junit.Assert;
+import org.testng.annotations.Test;
+
+import java.io.File;
+
+
+/**
+ * @author Anton_Mazur@epam.com, EPAM Systems, Inc.
+ *
+ * This test serve for verifying CRAMIterator records validation using strict validation strategy
+ **/
+
+public class CRAMIteratorTest extends HtsjdkTest {
+    private static final File refFile = new File("src/test/resources/htsjdk/samtools/cram/ce.fa");
+    private static final File cramFile = new File("src/test/resources/htsjdk/samtools/cram/ce#supp.3.0.cram");
+    ReferenceSource source = new ReferenceSource(refFile);
+
+    @Test
+    public void shouldNotThrowsExceptionIfCRAMfileContainsInvalidRecords() {
+        SAMRecordIterator cramIter =
+                getCramFileIterator(cramFile, source, ValidationStringency.STRICT);
+
+        Assert.assertTrue(cramIter.hasNext());
+    }
```

Comment:
This test needs a comment explaining what its testing (specifically, that validation is deferred until records are retrieved, rather than when cracking open the container), and a name that reflects that. Also, it would be helpful to reinforce that in the code by naming the iterator something like "cramIteratorOverInvalidRecords".